ports/mimxrt10xx/common-hal/microcontroller/Processor.c

```diff
@@ -50,6 +52,18 @@ float common_hal_mcu_processor_get_temperature(void) {
     return temp;
 }
 
+uint32_t common_hal_mcu_processor_set_frequency(mcu_processor_obj_t *self,
+    uint32_t frequency) {
+    uint32_t freq = frequency / 1000000;
+    if (freq != 24 && freq != 150 && freq != 396 && freq != 450 && freq != 528 && freq != 600 &&
+        freq != 720 && freq != 816 && freq != 912 && freq != 960 && freq != 1008) {
```

Comment:
This test is not exactly for the given frequencies, since the `/ 1000000` is truncating division. E.g. 24123456 will work for 24 MHz. Is that OK with you?

Reply:
No really should be integer value otherwise the test is going to be off - will have to fix that. Guess thats another one of those // divisions

Reply:
RIght, I didn't mean it was wrong, just that it accepts inexact frequencies and truncates them to the nearest valid frequency.